vm/systemSmartContracts/delegation.go

```diff
@@ -1039,13 +1097,16 @@ func (d *delegation) unDelegate(args *vmcommon.ContractCallInput) vmcommon.Retur
 		d.eei.AddReturnMessage("invalid value to undelegate")
 		return vmcommon.UserError
 	}
-	if activeFund.Value.Cmp(zero) > 0 && activeFund.Value.Cmp(d.minDelegationAmount) < 0 {
+
+	//TODO: The dust check shouldn't be done with the remained fund, subtracting valueToUndelegate from activeFund, and not directly with activeFund?
```

Comment:
the check is correct. the remaining funds should not be dust.

Reply:
Yes, but the original approach was different. So we should keep this new code added?